angular-app/src/app/app-routing.module.ts

```diff
@@ -1,55 +1,38 @@
 import { NgModule } from '@angular/core';
-import { RouterModule, Routes } from '@angular/router';
+import { Routes, RouterModule, UrlSegment, UrlMatchResult, RouteReuseStrategy } from '@angular/router';
 import { MainContentComponent } from "./components/main-content/main-content.component";
+import { AemPageDataResolver, AemPageRouteReuseStrategy } from "@adobe/cq-angular-editable-components";
 
-import { environment } from '../../src/environments/environment';
-
-const CONTEXT_PATH = environment.CONTEXT_PATH;
+export function AemPageMatcher(url: UrlSegment[]): UrlMatchResult {
+  if (url.length) {
```

Comment:
would "off" app link be caught by this as well? (ie. if you add a link to a page on the We.Retail Site)

Reply:
Thinking about this more, i wonder if this shouldnt be abstracted away into the NPM module.. so something like:

```
import { AemPageMatcher } from "@adobe/cq-angular-editable-components";

const routes = { 
   matcher: AemPageMatcher.create() // -> matches everything?
};

const routes = { 
   matcher: AemPageMatcher.create("^content/foo/.+\.html") // matches based on the pattern
};
```

Or something like that? (the API definition itself could/should be more elegant I imagine).

But I don't see why there isn't at least a provided default matcher if we think it will satisfy the 80% use case (which I imagine is any URL that isn't the root redirect?) It seems list most use-cases would need either a) a blank matcher or b) a regex matcher.

 This certainly wouldn't preclude custom matchers in more complex situations (since its use would be at the discretion of the developer creating the `app-routing.module.ts`.

Reply:
He also, due to the fact page model is not available, I would stick with the current form. Once Angular allows to use promise for resolver, we can enhance it by a lookup in the page model.